ui/app/components/pages/confirm-transaction-switch/confirm-transaction-switch.component.js

```diff
@@ -39,10 +41,15 @@ export default class ConfirmTransactionSwitch extends Component {
       return <Redirect to={{ pathname }} />
     }
 
-    if (fetchingMethodData) {
+    if (fetchingData) {
       return <Loading />
     }
 
+    if (!toSmartContract) {
```

Comment:
I'm thinking it might be better to have a prop of `isEther` or similar (maybe just a "`ether`" flag) where this `toSmartContract` seems to be the bulk of this component and this branch here seems to be the exception—thoughts?